### ISSUE TITLE
`ChatChannelHeaderView` is incorrectly hidden when the `ChatChannelVC`'s `channelController` doesn't initially have a `cid`

### ISSUE THREAD

**Issue by liamnichols** [GetStream/stream-chat-swift#2750]
## What did you do?

I presented an instance of `ChatChannelVC` using a `ChatChannelController` that was created using `channelController(createDirectMessageChannelWith:extraData:)`. 

## What did you expect to happen?

I expected the header to load, or to load shortly after presenting if data had to be fetched from the network. It should look like this:

<img src="https://github.com/GetStream/stream-chat-swift/assets/482871/f412e310-bc05-49a0-b63e-466be563962b" width="187.5" />

(taken after navigating from the channel list)

## What happened instead?

It never appears:

<img src="https://github.com/GetStream/stream-chat-swift/assets/482871/40a3025c-6d4b-4f61-94f0-6d9168cca8f5" width="187.5" />

(after using `channelController(createDirectMessageChannelWith:extraData:)`)

## GetStream Environment
**GetStream Chat version:** 4.35.2
**GetStream Chat frameworks:** StreamChat, StreamChatUI
**iOS version:** 16.4 (Simulator) and 16.6 (iPhone 12 Pro)
**Swift version:** 5.8
**Xcode version:** 14.3.1
**Device:** iPhone 12 Pro

## Additional context

I ended up getting confused so have given up, but it seems that when looking in the view debugger, the view is there, but with a zero frame and zero content:

<img width="1092" alt="Screenshot 2023-08-25 at 11 28 49" src="https://github.com/GetStream/stream-chat-swift/assets/482871/70d6772f-7165-421c-a97d-3b91e7d64793">

But I don't understand why it never gets populated with content. 

Similarly, the avatar never gets updated - the grey chefs had is the placeholder, both screenshots are for the same channel

**Comment by liamnichols** [comment]
Perhaps this:

https://github.com/GetStream/stream-chat-swift/blob/2a18b8c607f5ac6bcf89b52772e9023c4ea3f698/Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift#L183-L211

The `channelAvatarView.content` is only assigned the `channel` in `setUpLayout()`, but at this point in time, it's `nil` as the view controller's controller hasn't fetched it.

Similarly, `headerView.channelController` is only assigned if `channelController.cid != nil`, which it is because he `channelController` hasn't synchronised at this point in time.

Perhaps these should be updated in `ChatChannelVC.channelController(_:didUpdateChannel:)`?

**Comment by liamnichols** [comment]
I seem to have worked around this with a subclass:

```swift
// Workaround for https://github.com/GetStream/stream-chat-swift/issues/2750
final class ChatChannelVC: StreamChatUI.ChatChannelVC {
    override func channelController(
        _ channelController: ChatChannelController,
        didUpdateChannel channel: EntityChange<ChatChannel>
    ) {
        super.channelController(channelController, didUpdateChannel: channel)

        if headerView.channelController == nil, let cid = channelController.cid {
            headerView.channelController = client.channelController(for: cid)
        }

        if channelAvatarView.content.channel == nil {
            channelAvatarView.content = (channelController.channel, client.currentUserId)
        }
    }
}
```

**Comment by nuno-vieira** [comment]
Hi, @liamnichols!

Thank you for the report and for the debugging 👍 

We will be working on a fix soon, until then you can use your workaround 👍 

Best,
Nuno

**Comment by nuno-vieira** [comment]
Hi @liamnichols!

I can't reproduce this one either 🤔 

Can you show a snippet on how you are setting the channelController of `ChatChannelVC` ?

Best,
Nuno

**Comment by nuno-vieira** [comment]
Hi @liamnichols!

Thank you for all the details 👍 

This fix will be available on the next release 🙂 

Best,
Nuno

**Comment by nuno-vieira** [comment]
Hi!

This issue has been resolved on the new [4.37.0](https://github.com/GetStream/stream-chat-swift/releases/tag/4.37.0) Release! 🚀 

Best,
Nuno
